src/validations.ts

```diff
@@ -0,0 +1,92 @@
+import validator from "validator";
+import {
+	BankAccount,
+	DataModel,
+	SimplePayment,
+} from "./types.js";
+
+export enum ValidationErrorMessage {
+	InvalidIBAN = "Invalid IBAN",
+	InvalidBIC = "Invalid BIC",
+	InvalidCurrencyCode = "Invalid currency code",
+	InvalidDate = "Invalid date",
+}
+
+/**
+ * This error will be thrown in case of a validation issue. It provides message with error description and specific path to issue in dataModel object.
+ */
+export class ValidationError extends Error {
+	override name = "ValidationError";
+	path: string;
+
+	/**
+	 * @param message - explains, what is wrong on the specific field
+	 * @param path - navigates to the specific field in DataModel, where error occurred
+	 */
+	constructor(message: ValidationErrorMessage, path: string) {
+		super(String(message));
+		this.path = path;
+	}
+}
+
+/**
+ * validates bankAccount fields:
+ * - iban (ISO 13616)
+ * - bic (ISO 9362)
+ */
+
+export function validateBankAccount(bankAccount: BankAccount, path: string) {
+	if (!validator.isIBAN(bankAccount.iban)) {
+		throw new ValidationError(ValidationErrorMessage.InvalidIBAN, `${path}.iban`);
+	}
+	if (bankAccount.bic === "" || (bankAccount.bic && !validator.isBIC(bankAccount.bic))) {
+		throw new ValidationError(ValidationErrorMessage.InvalidBIC, `${path}.bic`);
+	}
+}
+
+/**
+ * validate simple payment fields:
+ * - currencyCode (ISO 4217)
+ * - paymentDueDate (ISO 8601)
+ * - bankAccounts
+ *
+ * @see validateBankAccount
+ */
+export function validateSimplePayment(simplePayment: SimplePayment, path: string) {
+	for (const [index, bankAccount] of simplePayment.bankAccounts.entries()) {
+		// todo: question is empty array of bank accounts valid input?
```

Comment:
It depends on the client application. Some may allow scanning with an empty target accounts if other payment info is given, while others may not. We need more research. I've created [issue](https://github.com/xseman/bysquare/issues/42) to check various banks, and we need to decide how many inputs to test. This could be a good test case.

Reply:
For now, we can skip checking for target accounts. We'll revisit this validation after more research.